core/templates/pages/collection-editor-page/services/collection-editor-state.service.ts

```diff
@@ -27,29 +27,30 @@ require('services/alerts.service.ts');
 
 require('pages/collection-editor-page/collection-editor-page.constants.ajs.ts');
 
+import { EventEmitter } from '@angular/core';
+
 angular.module('oppia').factory('CollectionEditorStateService', [
   '$rootScope', 'AlertsService', 'CollectionObjectFactory',
   'CollectionRightsBackendApiService', 'CollectionRightsObjectFactory',
   'EditableCollectionBackendApiService', 'UndoRedoService',
-  'EVENT_COLLECTION_INITIALIZED', 'EVENT_COLLECTION_REINITIALIZED',
   function(
       $rootScope, AlertsService, CollectionObjectFactory,
       CollectionRightsBackendApiService, CollectionRightsObjectFactory,
-      EditableCollectionBackendApiService, UndoRedoService,
-      EVENT_COLLECTION_INITIALIZED, EVENT_COLLECTION_REINITIALIZED) {
+      EditableCollectionBackendApiService, UndoRedoService) {
     var _collection = CollectionObjectFactory.createEmptyCollection();
     var _collectionRights = (
       CollectionRightsObjectFactory.createEmptyCollectionRights());
     var _collectionIsInitialized = false;
     var _collectionIsLoading = false;
     var _collectionIsBeingSaved = false;
+    var _collectionInitializedEventEmitter = new EventEmitter();
 
     var _setCollection = function(collection) {
       _collection.copyFromCollection(collection);
       if (_collectionIsInitialized) {
-        $rootScope.$broadcast(EVENT_COLLECTION_REINITIALIZED);
+        _collectionInitializedEventEmitter.emit();
```

Comment:
We are emitting the same event now unlike different events originally. Any reason for this?

Reply:
Across all uses, the events were behaving exactly the same, so I merged them into one event. I think this shouldn't be a problem, although the name does suggest it occurs only on initialization. Perhaps, I should change it to `_collectionRefreshedEventEmitter` (and the other refactoring similarly) instead?

Reply:
If both were behaving exactly similar -- I'm fine with replacing these two with one.